src/client/clientenvironment.cpp

```diff
@@ -519,9 +520,20 @@ void ClientEnvironment::getSelectedActiveObjects(
 			current_raw_normal = current_normal;
 		}
 		if (collision) {
-			current_intersection += obj->getPosition();
-			objects.emplace_back(obj->getId(), current_intersection, current_normal, current_raw_normal,
-				(current_intersection - shootline_on_map.start).getLengthSQ());
+			PointabilityType pointable;
+			if (pointabilities && ! gcao->isPlayer()) {
+				pointable = pointabilities->matchObject(gcao->getName(),
+							gcao->getGroups()).value_or(gcao->getProperties().pointable);
+			} else {
+				pointable = gcao->getProperties().pointable;
+			}
+			if (pointable != PointabilityType::POINTABLE_NOT) {
+				current_intersection += obj->getPosition();
+				objects.emplace_back(obj->getId(), current_intersection, current_normal, current_raw_normal,
+					(current_intersection - shootline_on_map.start).getLengthSQ(), pointable);
+			}
+			if (pointable == PointabilityType::POINTABLE_BLOCKING)
+				break;
```

Comment:
I don't think you can break here. `allObjects` is not sorted. (Also, `POINTABLE_BLOCKING` would block rays as much as normal `POINTABLE`.)
(Same for server environment.)

Reply:
fixed